index.bs

```diff
@@ -3049,6 +3051,14 @@ The [=remote end steps=] with |command parameters| are:
     1. Set the layout [[css2#viewport]] of the |context| to be |width| CSS pixels wide and |height| CSS pixels high.
 1. Else:
     1. Set the layout [[css2#viewport]] of |context| to the implementation-defined default layout viewport for |context|.
+1. Let |device pixel ratio parameter| be the |command parameters|["<code>devicePixelRatio</code>"] field.
+1. If |device pixel ratio parameter| is not null:
+  1. If the implementation is not able to set |device pixel ratio parameter| for |context| for any reasonm return [=error=] with [=error code=] [=unsupported operation=].
+  1. Override the [=device pixel ratio=] algorithm with the algorithm returning |device pixel ratio parameter| for |context|.
+  1. Perform implementation-specific steps to apply [=device pixel ratio=] to the page rendering for |context|.
+1. Else:
+  1. Restore the [=device pixel ratio=] algorithm for |context|.
```

Comment:
Restore to what?  I think it's OK to say something like `"Set the size of the [=CSS Pixel=] in |context|'s [=viewport=] to the implementation defined default."` or something like that.